packages/agent-utils/src/sys/workspaces/Workspace.ts

```diff
@@ -6,4 +6,6 @@ export interface Workspace {
   mkdirSync(subpath: string): void;
   readdirSync(subpath: string): string[];
   appendFileSync(subpath: string, data: string): void;
+  exec(command: string, args?: string[]): Promise<{ exitCode: number, stdout: string; stderr: string }>;
+  poetryInit(): Promise<void>;
```

Comment:
Why don't we use `exec` for this? It feels weird to have a use-case specific function within the `Workspace`'s definition.

Reply:
I moved the poetry logic to all be encapsulated in an agent function